cpp/src/slice2java/JavaUtil.cpp

```diff
@@ -616,7 +608,10 @@ Slice::computeDefaultSerialVersionUID(const ContainedPtr& p)
 
     const string data = os.str();
     long hashCode = 5381;
-    hashAdd(hashCode, data);
+    for (const auto& c : data)
```

Comment:
You should add a comment explaining why we don't use `std::hash<string>` here.

Reply:
Also, returning `long` doesn't sound right. It should be a well-defined integer like int64_t.

Reply:
We can probably use std::hash, but I think we should consider dropping serialization like we did in C#.

Reply:
I only inlined the logic, unfortunately I don't understand why it's written that way.

I did consider switching, but decided it's best not to break the existing `serialVersionUIDs` for no good reason.

Reply:
Sorry, I thought the answer was clear: we need our own hash function here because the generated code must be identical no matter which platform you compile on.

Reply:
Assuming the clearest answer is the correct answer has led me astray before...
I added a comment.